internal/grpc/services/sharesstorageprovider/sharesstorageprovider_test.go

```diff
@@ -237,6 +242,9 @@ var _ = Describe("Sharesstorageprovider", func() {
 				}
 			}, nil)
 
+		// reset share state and mount point
+		BaseShare.MountPoint = &sprovider.Reference{Path: "oldname"}
+		BaseShare.State = collaboration.ShareState_SHARE_STATE_ACCEPTED
```

Comment:
@aduffeck argh ... this should not be necessary, but the test modifies the BaseShare properties, which makes the tests interfere with each other ...

a deep copy might be one thing, or a factory function that we call in BeforeEach?

Reply:
@butonic just splitting declaration (in a `var` block) and initialization (in a `BeforeEach`/`JustBeforeEach`) should do the trick and ensure that each test gets a clean state to work with, e.g. 

```go
var (
    BaseShare *collaboration.ReceivedShare
)

BeforeEach(func() {
    BaseShare = &collaboration.ReceivedShare{}
})
```

See https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach for a more elaborate explanation.